src/solvers/refinement/string_refinement.cpp

```diff
@@ -227,8 +224,8 @@ bool string_refinementt::add_axioms_for_string_assigns(
     }
     else if(rhs.id()==ID_if)
     {
-      generator.add_axioms_for_if_array(lhs, to_if_expr(rhs));
-      return false;
+      add_symbol_to_symbol_map(lhs, rhs);
+      return true;
```

Comment:
I added some comments in the documentation of this function, but yes we need to clarify what is the behaviour of `set_to`